internal/webhook/external/refresh.go

```diff
@@ -69,6 +74,23 @@ func refreshWarehouses(
 		return lhs.Namespace == rhs.Namespace && lhs.Name == rhs.Name
 	})
 
+	if qualifier != "" {
+		var refreshEligibleWarehouses []kargoapi.Warehouse
```

Comment:
We can set the capacity to be equal to the total number of Warehouses in-hand so that we don't need to resize with every append.

```suggestion
		refreshEligibleWarehouses := make([]kargoapi.Warehouse, 0, len(warehouses))
```